libs/platform/src/lib/components/approval-flow/approval-flow.component.ts

```diff
@@ -496,223 +532,227 @@ export class ApprovalFlowComponent implements OnInit, OnDestroy {
     /** @hidden */
     _deleteCheckedNodes(): void {
         this._cacheCurrentApprovalProcess();
-        this._nodeComponents.filter(c => c._isSelected).forEach(({ node }) => {
-            this._deleteNode(node);
-            this._buildView(this._approvalProcess);
-        });
+
+        this._nodeComponents
+            .filter(c => c._isSelected)
+            .forEach(({ node }) => {
+                this._deleteNode(node);
+                this._buildView(this._approvalProcess);
+            });
```

Comment:
You could reduce the number of loops if you will make the check is the node selected in the `forEach`:

```suggestion
        this._nodeComponents
            .forEach(component => {
                if (component._isSelected) {
                    this._deleteNode(node);
                    this._buildView(this._approvalProcess);
                }
            });
```